packages/core/src/index.ts

```diff
@@ -223,6 +225,41 @@ export const evalEnergy = (s: State): number => {
   return objective(weight)(s.varyingValues);
 };
 
+/**
+ * Evaluate a list of constraints/objectives: this will be useful if a user want to apply a subset of constrs/objs on a `State`
+ * TODO: comment this properly
+ * NOTE: The type has to be passed in because otherwise we can't distinguish between the kinds of functions
+ */
+export const evalFns = (fns: Fn[], s: State): number[] => {
+  const { objFnCache, constrFnCache } = s.params;
+
+  // NOTE: if `prepareState` hasn't been called before, log a warning message and generate a fresh optimization problem
+  if (!objFnCache || !constrFnCache) {
+    log.warn(
+      "State is not prepared for energy evaluation. Call `prepareState` to initialize the cached objective/constraint functions first."
+    );
+    const newState = genFns(s);
```

Comment:
Can this be extended to allow not just a subset of existing obj/constrs? Say the user constructs a new constraint and passes it as a part of `fns`. Is `genFn` enough for including this new constraint in the opt problem?

Reply:
No, it only generates the function for that individual function. If you want to include a new constraint/objective in the overall opt problem (including having an overall gradient), it will need to be properly weighted and added to the rest of the evaluated functions, so you should regenerate the whole function using `evalEnergyOnCustom`. 

I suppose if you really wanted to, you could compile the individual function and evaluate it, then do the same weighting-adding process to its results using pure JS math, and then manually change the gradient to include the individual function ... but that seems very prone to bugs.